### PR TITLE
[graphqlErrorHandler] Fix us reporting all errors but syntax errors.

### PR DESCRIPTION
**Pull request by alloy**
Saw that we were reporting these 404s again:

* https://sentry.io/artsynet/metaphysics-production/issues/810021918/?query=is%3Aunresolved
* https://sentry.io/artsynet/metaphysics-production/issues/873514466/?query=is%3Aunresolved

### CODE REVIEW

File: src/lib/graphqlErrorHandler.ts [hunk 1]
```diff
@@ -125,7 +120,7 @@ export const graphqlErrorHandler = (
     const flattenedErrors = flattenErrors(topLevelError)
     if (enableSentry) {
       flattenedErrors.forEach(e => {
-        if (shouldReportError(e) || shouldReportParentError(e.originalError)) {
```
Comment (alloy) [inline src/lib/graphqlErrorHandler.ts]: What happened was that for any error other than syntax errors, `shouldReportError` would return `true` and thus never perform the checks of `shouldReportParentError`. I now rolled that into one.

File: src/lib/graphqlErrorHandler.ts [hunk 2]
```diff
@@ -34,22 +34,17 @@ interface QueryContext {
...
-export const shouldReportParentError = (
-  error: null | undefined | Error | HTTPError
+  error: null | undefined | Error | HTTPError | GraphQLError
 ) => {
   if (error) {
+    if (error instanceof GraphQLError) {
+      if (error.message.startsWith("Syntax Error")) {
+        return false
+      } else {
+        return shouldReportError(error.originalError)
```
Comment (alloy) [inline src/lib/graphqlErrorHandler.ts]: recursive-kittens.gif

File: src/lib/graphqlErrorHandler.ts (hunk 1 above)
Comment (orta) [reply src/lib/graphqlErrorHandler.ts]: ace

**Comment by orta** [comment]
Great -yep

**Pull request merged**